hangman.py

```diff
@@ -0,0 +1,196 @@
+
+
+def letter_in_word(secret_word,letter,letters_a):
+    """
+    Check if letter is word, else subtracts point
+    """
+    global letter_guessed
+    global guesses_point
+    letters_guessed.append(letter)
+    if letter in secret_word:
+        print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+    else:
+        if letter in letters_a:
+            guesses_point -= 2
+        else:
+            guesses_point -= 1
+        print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+
+
+def hangman(secret_word, request):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+    '''
+    global letters_guessed
+    global warnings_point
+    global guesses_point
+    letters_a = frozenset("aeiou")
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+    while guesses_point > 0:
+        print("-------\nYou have", guesses_point, "guesses left\nAvailable letters:", end=" ")
+        print(get_available_letters(letters_guessed))
+        letter = input("Please guess a letter:")
+        if request == "hints" and letter=="*":
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+        else:
+            if str.isalpha(letter.lower()) == False or len(letter) != 1:
+                warnings("That is not a valid letter or you print more than one letter.")
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter in letters_guessed:
+                warnings("You have already guessed that letter:")
+                print(get_guessed_word(secret_word, letters_guessed))
+            else:
+                letter_in_word(secret_word, letter, letters_a)
+        if is_word_guessed(secret_word, letters_guessed) == 0:
+            final_points = guesses_point * len(set(secret_word))
+            print("------\nCongratulations, you won!\nWord:",secret_word,"\nYour total score for this game is:", final_points)
+            guesses_point = -1
+        elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+            print("------\nSorry, you ran out of guessed. The word was", secret_word)
+            break
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    word = my_word.replace(" ", "")
```

Comment:
The `my_word` is in most cases the same, so I'd recommend you to avoid many times replacement of the same word